__test__/lib/fifo_animal_shelter.test.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const AnimalShelter = require('../../lib/fifo_animal_shelter.js');
+
+class Animal {
+  constructor(type) {
+    this.type = type;
+  }
+}
+
+describe('enqueue(animal)', () => {
+
+  it('enqueue(animal) creates a new array object', () => {
+    let test = new AnimalShelter();
+      expect(typeof test.queue).toEqual('object');
+  });
+
+  it('enqueue(animal) puts the new animal at the back of the queue (end of array)', () => {
+    let test = new AnimalShelter();
+    test.enqueue('dog');
+    test.enqueue('cat');
+    expect(test.queue[1].type).toEqual('cat');
+  });
+
+  it(`dequeue(pref) returns the first animal in the queue indicated by pref `, () => {
```

Comment:
This test currently is not passing - take a look and see if you can figure out why